namerd/storage/zk/src/main/scala/com/twitter/finagle/serverset2/ZkDtabStore.scala

```diff
@@ -44,6 +44,22 @@ class ZkDtabStore(
     children <- zk.getChildrenOf(zkPrefix)
   } yield children.children.toSet
 
+  private[this] def ensurePath(zkw: ZooKeeperWriter, path: String) {
+    val components = Path.read(path)
+    if (components.size > 1) {
+      ensurePath(zkw, components.take(components.size - 1).show)
+    }
+
+    zkw.create(
+      path,
+      None,
+      Seq(Data.ACL.AnyoneAllUnsafe),
+      CreateMode.Persistent
+    ).rescue {
+        case KeeperException.NodeExists(_) => Future.Unit
+      }.unit
```

Comment:
better:
```scala
...unit.handle { case KeeperException.NodeExists(_) => }
```

i'm a little surprised this compiles given that create returns one type and rescue returns another...